sale_blanket_orders/views/blanket_orders.xml

```diff
@@ -0,0 +1,121 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+    <record id="view_blanket_order_tree" model="ir.ui.view">
+        <field name="name">sale.blanket.order.tree</field>
+        <field name="model">sale.blanket.order</field>
+        <field name="arch" type="xml">
+            <tree>
+                <field name="name"/>
+                <field name="partner_id"/>
+                <field name="validity_date"/>
+                <field name="state"/>
+            </tree>
+        </field>
+    </record>
+
+    <record id="view_blanket_order_form" model="ir.ui.view">
+        <field name="name">sale.blanket.order.form</field>
+        <field name="model">sale.blanket.order</field>
+        <field name="arch" type="xml">
+            <form string="Blanket Order" class="o_sale_order">
+                <header>
+                    <button name="%(action_create_sale_order)d" string="Create Sale Order"
+                        type="action" class="btn-primary"
+                        attrs="{'invisible': [('state', '!=', 'opened')]}"/>
+                    <button name="action_confirm" states="draft" string="Confirm" class="btn-primary o_sale_confirm" type="object" />
+                    <field name="state" widget="statusbar" statusbar_visible="draft,opened,expired"/>
+                </header>
+                <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button name="action_view_sale_orders"
+                            type="object"
+                            class="oe_stat_button"
+                            icon="fa-pencil-square-o">
+                            <field name="sale_count" widget="statinfo" string="Sale Orders"/>
+                        </button>
+                    </div>
+                    <div class="oe_title">
+                        <h1>
+                            <field name="name" readonly="1"/>
+                        </h1>
+                    </div>
+                    <group>
+                        <group>
+                            <field name="partner_id" domain="[('customer','=',True)]" context="{'search_default_customer':1, 'show_address': 1}" options='{"always_reload": True}'/>
+                        </group>
+                        <group>
+                            <field name="validity_date"/>
```

Comment:
should be set required in draft state through an attrs attribute

Reply:
No problem. I had added it initially, but then thought it could be useful to create the order before the validity date is agreed, so instead I added a check when the user confirms it.
I'll add them back.